### ISSUE TITLE
Add OneAnd tests

### ISSUE THREAD

**Issue by ceedubs** [non/cats#255]
At a minimum we can check laws for the instances it provides.

It's probably best to wait until #254 is merged before starting this work.

**Comment by lvicentesanchez** [comment]
I will deal with this one if nobody has already started.

**Comment by ceedubs** [comment]
Thanks @lvicentesanchez! As far as I know, nobody has started this work.

**Comment by non** [comment]
Is this still in progress?

**Comment by lvicentesanchez** [comment]
Sorry I was trapped by "real world" stuff. I will work on this the next couple of days and submit a PR.

**Comment by non** [comment]
No problem, we are all volunteers here! :)

I just wanted to "unclaim" the bug if you were not going to be able to get to it.

**Comment by adelbertc** [comment]
What's the status on this? @lvicentesanchez if you're still trapped by stuff I can take this up.

**Comment by lvicentesanchez** [comment]
yes. you should take it up I you can... I haven't had time to look at this
yet :(

**Comment by adelbertc** [comment]
I got this.

**Comment by adelbertc** [comment]
To test the different instances with more general constraints (https://github.com/non/cats/blob/master/core/src/main/scala/cats/data/OneAnd.scala#L80-119), I was thinking of having something like this, possibly in `cats-tests`.

```scala
object TestData {
  final case class ListWrapper[A](list: List[A]) extends AnyVal

  object ListWrapper {
    def foldable: Foldable[ListWrapper] = . . .
    def functor: Functor[ListWrapper = . . .
    // etc...
  }
}
```

Note the lack of `implicit` on the type class instances - this would allow us to test each instance by wrapping the tests in it's own block with an implicit val assigned to the type class instance we want in scope. What do people think?

**Comment by ceedubs** [comment]
Fixed by #346
